### PR TITLE
Logistic regression

### LINKED ISSUE #5: Logistic Regression
Имплементиране на методите и функциите за логистична регресия

# Структура на логистичната регресия
- [x] Структура с коефициенти и указатели към методите на алгоритъма
- [x] Сигмоидна функция приемаща за вход резултат от линейна регресия
image
Сигмоидна функция
image
Енкапсулирана линейна регресия(В случая множествена линейна регресия)
- [x] Loss function, фунцкия калкулираща грешката на предсказания, от алгоритъма, резултат от реалния (За едно поле от тренировъчните данни)
image
->
image

- [x] Cost function, функция калкулираща грешката на модела върху целите тренировъчни данни
image

- [x] Optimizer function - Gradient descent, оптимизираща функция 
image
Gradient descent, very scary math
- [x] Fit function, функция която тренира алгоритъма върху въведените данни
- [x] Predict function, функция която взима коефициентите и връща предсказания резултат спрямо въведените данни на базата на тренираните коефициенти

# Ресурси за справка:
- [Logistic regression from scratch](https://philippmuens.com/logistic-regression-from-scratch)
- [Gradient descent](https://philippmuens.com/gradient-descent-from-scratch)
- [Logistic regression from scratch(Medium)](https://towardsdatascience.com/coding-logistic-regression-from-scratch-d18b4fbfca8a)
- [Logistic regression intuition (Medium)](https://medium.com/analytics-vidhya/basics-and-beyond-logistic-regression-34549d2ee800)

### PR DESCRIPTION
**Pull request by CodenameSource**
Work in progress (see [issue](https://github.com/CodenameSource/libuaic/issues/5))

### CODE REVIEW

**Comment by CodenameSource** [comment]
TODO: Clean code and fix memory leaks(stray mallocs in fit, silly me)

**Comment by CodenameSource** [comment]
God have mercy on my soul for merging such a garbage fire into main
